roseau/load_flow/models/loads/loads.py

```diff
@@ -305,86 +369,6 @@ def to_dict(self) -> JsonDict:
-
-            parameters:
-                List of flexible parameters for each phase.
-
-            phases:
-                The phases of the load. A string like ``"abc"`` or ``"an"`` etc. The order of the
-                phases is important. For a full list of supported phases, see the class attribute
-                :attr:`allowed_phases`. All phases of the load, except ``"n"``, must be present in
-                the phases of the connected bus. By default, the phases of the bus are used.
-        """
-        super().__init__(id=id, phases=phases, bus=bus, s=s, **kwargs)
-        if len(parameters) != 3:
-            msg = f"Incorrect number of parameters: {len(parameters)} instead of 3"
-            logger.error(msg)
-            raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_PARAMETERS_SIZE)
-
-        for power, parameter in zip(self.s, parameters):
-            if abs(power) > parameter.s_max and (
-                parameter.control_p.type != "constant" or parameter.control_q.type != "constant"
-            ):
-                msg = f"The power is greater than the parameter s_max for flexible load {self.id!r}"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_S_VALUE)
-            if parameter.control_p.type == "p_max_u_production" and power.real > 0:
-                msg = f"There is a production control but a positive power for flexible load {self.id!r}"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_S_VALUE)
-            if parameter.control_p.type == "p_max_u_consumption" and power.real < 0:
-                msg = f"There is a consumption control but a negative power for flexible load {self.id!r}"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_S_VALUE)
-            if parameter.control_p.type != "constant" and power.real == 0:
-                msg = f"There is a P control but a null active power for flexible load {self.id!r}"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_S_VALUE)
-
-        self.parameters = parameters
-        self._powers = None
-
-    @property
-    @ureg.wraps("VA", None, strict=False)
-    def powers(self) -> np.ndarray:
-        """An array of the actual powers of each phase (VA) as computed by the load flow."""
-        return self._powers
-
-    @powers.setter
-    def powers(self, value: np.ndarray):
-        self._powers = value
-
-    def to_dict(self) -> JsonDict:
-        return {**super().to_dict(), "parameters": [p.to_dict() for p in self.parameters]}
-
-
 AbstractLoad._power_load_class = PowerLoad
 AbstractLoad._current_load_class = CurrentLoad
 AbstractLoad._impedance_load_class = ImpedanceLoad
-AbstractLoad._flexible_load_class = FlexibleLoad
```

Comment:
I think this last line should be replaced with `AbstactLoad._flexible_parameter_class = FlexibleParameter`. (Here or somewhere else)